content/utilities/biome-heatmap/plugin.js

```diff
@@ -0,0 +1,346 @@
+
+// Two modes to choose from:
+// 1. Draw SpaceType (can show you space type in not mined areas of the map)
+// 2. Draw Biome heatmap (draws biome perlin value that together with the spacetype define the biome)
+
+// Resource intensive plugin - dont zoom out too much
+// Hover over text inside the plugin to get a tooltip
+
+const quadrantSize = 500; // smaller size means more detail but also longer computation
+
+const biomeColorAccuracy = 3; // smaller = more accurate & more lag, 1 is best accuracy
+
+const spaceTypeOpacity = "88"; // from 00 to FF
+const biomeOpacity = "0.4"; // from 0 to 1
+
+// useful package containing types (its not importet because color values also need to change if types change)
+// import("https://cdn.skypack.dev/@darkforest_eth/types").then(console.log)
+
+let spaceType = {};
+spaceType.NEBULA = 0;
+spaceType.SPACE = 1;
+spaceType.DEEP_SPACE = 2;
+spaceType.DEAD_SPACE = 3;
```

Comment:
the types package contains a `SpaceType` const enum which you could use instead of declaring the constant here. I don't forsee us adding space types any time in the near future so this should be fine for now.

https://github.com/darkforest-eth/packages/blob/master/types/src/game_types.ts#L11-L16

Reply:
thanks for pointing it out, if another space type gets added to the game i will update the plugin so that it uses the package